src/pin.ts

```diff
@@ -51,120 +62,106 @@ class ContractPinner {
+    } else {
+      // Create the new version directory structure
+      await this.createVersionDirectory(contractName, version)
     }
 
-    // Create the new version directory structure
-    await this.createVersionDirectory(contractName, version)
-
-    // Copy contract files to the new version directory
-    await this.copyContractFiles(contractPath, contractName, version)
+    // Copy contract files (this will handle --overwrite and --only-changed logic)
+    await this.copyContractFiles(manifestPath, contractFiles, contractName, version)
 
     // Update chelonia.json configuration
-    await this.updateCheloniaConfig(contractName, version, contractPath)
-
-    // Note: Manifests should be generated using 'chel manifest' command
-    // The pin command only copies existing manifests if they exist
+    await this.updateCheloniaConfig(contractName, version, manifestPath)
 
     console.log(colors.green(`✅ Successfully pinned ${contractName} to version ${version}`))
     console.log(colors.gray(`Location: contracts/${contractName}/${version}/`))
   }
 
   /**
-   * Load chelonia.json configuration file
-   * Creates a default one if it doesn't exist
+   * Parse manifest file to extract contract information
+   * @param manifestPath - Path to the manifest file
+   * @returns Object containing contract name and file information
    */
-  private async loadCheloniaConfig () {
-    const configPath = join(this.projectRoot, 'chelonia.json')
+  private async parseManifest (manifestPath: string): Promise<{ contractName: string, contractFiles: ContractFiles }> {
+    try {
+      const manifestContent = await readFile(manifestPath, 'utf8')
+      const manifest = JSON.parse(manifestContent)
+      const body = JSON.parse(manifest.body)
 
-    if (existsSync(configPath)) {
-      try {
-        const configContent = await readFile(configPath, 'utf8')
-        this.cheloniaConfig = JSON.parse(configContent)
-        console.log(colors.blue('📄 Loaded existing chelonia.json'))
-      } catch (error) {
-        console.warn(colors.yellow(`Warning: Could not parse chelonia.json: ${error}`))
-        // Use default empty config
+      const fullContractName = body.name
+      const mainFile = body.contract.file
+      const slimFile = body.contractSlim?.file
+
+      if (!fullContractName || !mainFile) {
+        throw new Error('Invalid manifest: missing contract name or main file')
       }
-    } else {
-      console.log(colors.blue('📄 Creating new chelonia.json'))
-      // Will be saved when updateCheloniaConfig is called
-    }
 
-    // Ensure contracts object exists
-    if (!this.cheloniaConfig.contracts) {
-      this.cheloniaConfig.contracts = {}
-    }
-  }
+      // Extract just the contract name part (after the last slash)
+      // e.g., "gi.contracts/group" -> "group"
+      const contractName = fullContractName.split('/').pop() || fullContractName
```

Comment:
Not sure about this. @taoeffect ? I believe using the full name is fine and even desirable.

Reply:
https://github.com/okTurtles/chel/pull/64#discussion_r2270522755